src/main/java/org/opensearch/ad/transport/SearchAnomalyResultTransportAction.java

```diff
@@ -73,26 +73,41 @@ public SearchAnomalyResultTransportAction(
     @Override
     protected void doExecute(Task task, SearchRequest request, ActionListener<SearchResponse> listener) {
         String[] indices = request.indices();
+        if (indices == null || indices.length == 0) {
+            listener.onFailure(new IllegalArgumentException("No indices set in search request"));
+            return;
+        }
         // Set query indices as default result indices, will check custom result indices permission and add
         // custom indices which user has search permission later.
-        request.indices(ALL_AD_RESULTS_INDEX_PATTERN);
 
-        Set<String> customResultIndices = new HashSet<>();
-        if (indices != null && indices.length > 0) {
-            String[] concreteIndices = indexNameExpressionResolver
-                .concreteIndexNames(clusterService.state(), IndicesOptions.lenientExpandOpen(), indices);
-            if (concreteIndices == null || concreteIndices.length == 0) {
-                // No custom result indices found, just search default result index
-                searchHandler.search(request, listener);
-                return;
+        boolean onlyQueryCustomResultIndex = true;
+        for (String indexName : indices) {
+            // If only query custom result index, don't need to set ALL_AD_RESULTS_INDEX_PATTERN in search request
+            if (ALL_AD_RESULTS_INDEX_PATTERN.equals(indexName)) {
+                onlyQueryCustomResultIndex = false;
             }
-            for (String index : concreteIndices) {
-                if (index.startsWith(CUSTOM_RESULT_INDEX_PREFIX)) {
-                    customResultIndices.add(index);
-                }
+        }
+
+        String[] concreteIndices = indexNameExpressionResolver
+            .concreteIndexNames(clusterService.state(), IndicesOptions.lenientExpandOpen(), indices);
+        if (concreteIndices == null || concreteIndices.length == 0) {
+            // No custom result indices found, just search default result index
```

Comment:
Comment here seems confusing. Since `concreteIndices` can include the default result system index, won't this `if` block only be true if no indices in the entire cluster match the search request's `indices`?

Reply:
Good catch, will tune the comment, should be 
"// No result indices found, will throw exception"